compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -513,7 +516,7 @@ pub(super) fn thir_abstract_const<'tcx>(
         let body = tcx.thir_body(def);
         if body.0.borrow().exprs.is_empty() {
             // type error in constant, there is no thir
-            return Err(ErrorReported);
+            return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
Whatever type `thir_body` returns should contain an `ErrorReported` in the error case.